docs/learn/documentation/versioned/jobs/samza-configurations.md

```diff
@@ -77,9 +77,10 @@ These are the basic properties for setting up a Samza application.
+|job.systemstreampartition.<br>matcher.config.<br>range| |If `job.systemstreampartition.matcher.class` is specified, and the value of this property is `org.apache.samza.system.RangeSystemStreamPartitionMatcher`, then this property is a required configuration. Specify a comma separated list of range(s) to determine which partition matches, and thus statically assigned to the Job. For example "2,3,11-20", statically assigns partition 2, 3, and 11 to 20 for all the specified system and streams (topics in case of Kafka) to the job. A single configuration value like "19" is valid as well. This statically assigns partition 19. For config validation each element in the comma separated list much conform to one of the following regex:<br>"`(\\d+)`" or "`(\\d+-\\d+)`"|
 |job.systemstreampartition.<br>matcher.config.<br>regex| |If `job.systemstreampartition.matcher.class` is specified, and the value of this property is `org.apache.samza.system.RegexSystemStreamPartitionMatcher`, then this property is a required configuration. The value should be a valid Java supported regex. For example "[1-2]", statically assigns partition 1 and 2 for all the specified system and streams (topics in case of Kakfa) to the job.|
 |job.systemstreampartition.<br>matcher.config.<br>job.factory.regex| |This configuration can be used to specify the Java supported regex to match the StreamJobFactory for which the static partition assignment should be enabled. This configuration enables the partition assignment feature to be used for custom StreamJobFactory(ies) as well.<br>This config defaults to the following value: "_org\\\\.apache\\\\.samza\\\\.job\\\\.local(.*ProcessJobFactory &#124; .*ThreadJobFactory)_", which enables static partition assignment when job.factory.class is set to `org.apache.samza.job.local.ProcessJobFactory` or `org.apache.samza.job.local.ThreadJobFactory`.|
+|job.systemstreampartition.<br>grouper.proxy.enabled|true|When enabled, this provides a proxy layer on top of the grouper configured by `job.systemstreampartition.grouper.factory` that allows stateful jobs to expand or contract their partition count by a multiple of the previous count so that events from an input stream partition are not stored in the local state of a different task. This will prevent erroneous results. If this configuration is disabled or if the job is stateless, the proxy will call the grouper directly and skip the expansion or contraction logic. See [SEP-5](https://cwiki.apache.org/confluence/display/SAMZA/SEP-5%3A+Enable+partition+expansion+of+input+streams) for more details.|
```

Comment:
Can we rename this config to reflect the feature (partition count increase) instead of the implementation (grouper proxy)? Same for the docs.

Reply:
Sure will change.